handler/handler.go

```diff
@@ -264,17 +273,20 @@ func (h *ReindexRequestedHandler) getAndSendDatasetURLsFromLatestMetadata(ctx co
 	return datasetURLChan
 }
 
-// TODO - logExtractedDatasetURLs is temporary.
-// The dataset url should be sent to the content-updated topic here in the future.
-// But for the time being, we are going to extract the urls and print them
-func (h *ReindexRequestedHandler) logExtractedDatasetURLs(ctx context.Context, wgDataset *sync.WaitGroup, cfg *config.Config, datasetURLChan chan string, reindexReqEvent *models.ReindexRequested) int {
+func (h *ReindexRequestedHandler) sendExtractedDatasetURLs(ctx context.Context, wgDataset *sync.WaitGroup, cfg *config.Config, datasetURLChan chan string, reindexReqEvent *models.ReindexRequested) int {
 	var urlListCount int
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
 	go func() {
+		defer wg.Done()
```

Comment:
No need to add another WaitGroup. Instead, remove the go routine (`go func…`) from this function entirely including the deferred wg.Dones and instead make the call sequential. It's what it is doing effectively anyway, just making it explicit.
```
func (h *ReindexRequestedHandler) sendExtractedDatasetURLs(ctx context.Context, wgDataset *sync.WaitGroup, cfg *config.Config, datasetURLChan chan string, reindexReqEvent *models.ReindexRequested) int {
	var urlListCount int
	for datasetURL := range datasetURLChan {
		log.Info(ctx, "send extracted dataset urls")
		err := h.ContentUpdatedProducer.ContentUpdate(ctx, cfg, models.ContentUpdated{
			URI:         datasetURL,
			DataType:    DatasetDataType,
			JobID:       reindexReqEvent.JobID,
			TraceID:     reindexReqEvent.TraceID,
			SearchIndex: reindexReqEvent.SearchIndex,
		})
		if err != nil {
			log.Error(ctx, "failed to publish datasets to content update topic", err)
			break
		}
		urlListCount++
	}
	return urlListCount
}
```
NB. Had to replace a return with a break as it's no longer in a goroutine. The error logic isn't great anyway, but this works better.

Reply:
@jessjenkins finally got around to finishing this, if you've got a sec for a re-review.